File: Longest_Common_Prefix.cc
/*
Write a function to find the longest common prefix string amongst an array of strings.
*/

class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        if (strs.empty())
            return "";
            
        for (int i = 0; i < strs[0].size(); i++)
        {
            for (int j = 1; j < strs.size(); j++)
            {
                if (strs[0][i] != strs[j][i])
                {
                    //return strs[0].substr(0, i);
                    //return string(strs[0].c_str(), i);
                    return string(strs[0], 0, i);
                }
            }
        }
        
        return strs[0];
    }
};

File: go/longest-common-prefix/longest-common-prefix.go
package longest_common_prefix

/*
Write a function to find the longest common prefix string amongst an array of strings.
*/


// 或者,先找出最短的str
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }

    prefix := make([]byte, 0, len(strs[0]))
    s := []byte(strs[0])
    for i, c := range s {
        for j := 1; j < len(strs); j++ {
                goto exit
            }
        }
        prefix = append(prefix, c)
    }

exit:
    return string(prefix)
}
